agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -260,23 +260,17 @@ public class OpenTelemetry {
 
                 val inferenceSpan = spanProcessor.getSpanOrThrow<InferenceSpan>(inferenceSpanId)
 
-                val provider = eventContext.model.provider
-
-                // Add events to the InferenceSpan before finishing the span
-                val lastMessage = eventContext.responses.lastOrNull()
-
-                val events: List<GenAIAgentEvent> = lastMessage?.let { message ->
-                    buildList {
-                        when (message) {
-                            is Message.Assistant -> add(ChoiceEvent(provider, message, config.isVerbose))
-                            else -> {}
-                        }
+                eventContext.prompt.messages
+                    .filter {
+                        // TODO: check tool message here
+                        it.role == Message.Role.Assistant
+                    }
+                    .forEachIndexed { index, message ->
+                        inferenceSpan.span.setAttribute("gen_ai.completion.$index.role", message.role.name.lowercase())
```

Comment:
@agbragin , here as well. Please refer to the OT Semantic convention doc (https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-spans/#inference)